### PR TITLE
Homework for fourth lecture

### PR DESCRIPTION
**Pull request by Khazel**
Completed homework for lecture four. Added .idea folder to the .gitignore file.

### CODE REVIEW

File: src/Lecture_4/Homework_4/Task_11.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+public class Task_11 {
+
+    public static void main(String[] args) {
+
+        Scanner sn = new Scanner(System.in);
+
+        System.out.println("Please enter an email");
+        String input = sn.nextLine();
+
+        Pattern pattern = Pattern.compile("^[A-Za-z0-9+_.-]+@(.+)$");
```
Comment (KaloyanTsvetkov) [inline src/Lecture_4/Homework_4/Task_11.java]: Based on your regex this are valid emails:
bl....a@abv.bggggggg
bl....a@abv
bl@@@@@a@abv.

File: src/Lecture_4/Homework_4/Task_11.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        Matcher matcher = pattern.matcher(input);
+
+        if (matcher.matches()) {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_4/Homework_4/Task_11.java:19]: try using:
^(?=.{1,64}@)[A-Za-z0-9_-]+(\.[A-Za-z0-9_-]+)*@[^-][A-Za-z0-9-]+(\.[A-Za-z0-9-]+)*(\.[A-Za-z]{2,})$

**Review by KaloyanTsvetkov — APPROVED** [review]

File: src/Lecture_4/Homework_4/Task_11.java (hunk 1 above)
Comment (Khazel) [reply src/Lecture_4/Homework_4/Task_11.java]: The last one looks especially tricky.

File: src/Lecture_4/Homework_4/Task_11.java (hunk 2 above)
Comment (Khazel) [reply src/Lecture_4/Homework_4/Task_11.java:19]: It works much better. I will spend more time looking at regex topic.

**Pull request merged**